signer/src/bin/main.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Offline transaction signer implementation
+//!
+//! WIP port / simplification from https://github.com/mobilecoinofficial/full-service/blob/fefe6f645d676b393ece2f607f0081304141b590/transaction-signer/src/bin/main.rs#L337
+
+use std::path::Path;
+
+use bip39::{Language, Mnemonic, MnemonicType};
+use clap::Parser;
+use log::{debug, info};
+use serde::{Deserialize, Serialize};
+
+use mc_core::{account::Account, slip10::Slip10KeyGenerator};
+use mc_crypto_ring_signature_signer::LocalRingSigner;
+use mc_transaction_core::AccountKey;
+use mc_transaction_signer::{read_input, write_output, Operations};
+
+#[derive(Clone, PartialEq, Debug, Parser)]
+struct Args {
+    /// Account secrets file
+    #[clap(long, short, default_value = "mc_secrets.json")]
+    secret_file: String,
+
+    #[command(subcommand)]
+    action: Actions,
+}
+
+#[derive(Clone, PartialEq, Debug, Parser)]
+enum Actions {
+    /// Create a new offline account, writing secrets to the output file
+    Create {
+        /// File name for account secrets to be written to
+        #[clap(short, long)]
+        output: String,
+    },
+    /// Import an existing offline account via mnemonic
+    Import {
+        /// Optional account name
+        #[clap(short, long)]
+        name: Option<String>,
+
+        /// Mnemonic for account import
+        mnemonic: String,
```

Comment:
Shouldn't there be a `#[clap]` attribute here?
Separately, having the mnemonic passed as a command line argument is a very good way of having it get stored in your shell history, which is not great... 
It should at the very least be possible to pass it via an environment variable